### PR TITLE
Experience new multi

### PR DESCRIPTION
**Pull request by sashakaverina**
- implemented multi-step form for creating a new experience (https://github.com/zombocom/wicked/wiki/Building-Partial-Objects-Step-by-Step)
 - status property for experience model to separate ready and in progress experiences
 - progress bar for steps
 - footer for buttons 'back' and 'next'

need to work on:
- simple_form validations aren't working since I need to create an instance before the form. how to validate on each step?
- how to redirect to #show page after finish
- map isn't showing
- add style

### CODE REVIEW

**Comment by sashakaverina** [comment]
Screen Shot 2022-01-08 at 16 18 45

File: app/assets/stylesheets/components/_map.scss [hunk 1]
```diff
@@ -114,4 +114,15 @@
...
+#wrapper { 
+  position: relative;
+}
+
+#over_map { 
+  position: absolute; 
+  top: 35px; 
+  left: 35%; 
+  z-index: 99;
 }
```
Comment (dmbf29) [inline app/assets/stylesheets/components/_map.scss:128]: What's this for?

File: app/controllers/experiences_controller.rb [hunk 2]
```diff
@@ -19,5 +20,13 @@ def show
   end
 
   def new
+    @experience = Experience.new
+    @experience.user = current_user
+    @experience.status = "building"
```
Comment (dmbf29) [inline app/controllers/experiences_controller.rb:25]: if you have `default: 'building'` in your migration then you wouldn't need this line

File: app/views/experiences/new.html.erb [hunk 3]
```diff
@@ -1 +1,30 @@
...
+          <a class="btn btn-primary mt-3 smooth" href="#">Next</a>
+      </div>
+
+    </div>
+  </div>
+</div>
+
+
+<% end %>
+
```
Comment (dmbf29) [inline app/views/experiences/new.html.erb]: Are you using this file anymore?

File: db/migrate/20220108025128_add_status_to_experiences.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToExperiences < ActiveRecord::Migration[6.1]
+  def change
+    add_column :experiences, :status, :string
```
Comment (dmbf29) [inline db/migrate/20220108025128_add_status_to_experiences.rb:3]:
```
default: 'building
```

File: app/views/experiences/build/add_price.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<%= simple_form_for @experience, url: wizard_path, method: :put do |f| %>
+  <div class="container">
+    <div class="row justify-content-center mt-3">
+      <div class="col-12 col-sm-10 col-md-8 col-lg-6">
+        <h2 class="mt-2">How much do you charge for your experience?</h2>
+        <p class="mt-5">Make sure you include all the fees. You can always specify additional fees later.  </p>
+        <%= f.input :price, as: :integer, :input_html => {:rows => 6}, html5: true, label: false, placeholder: "5000 JPY" %>
```
Comment (dmbf29) [inline app/views/experiences/build/add_price.html.erb:7]: :trollface: 
```
input_html: { rows: 6 }
```

File: app/views/experiences/build/_sidebar.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  <li class="nav-item">
+    <a class="nav-link" href="#">Link</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" href="#">Link</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link disabled" href="#">Disabled</a>
+  </li>
+</ul>
```
Comment (dmbf29) [inline app/views/experiences/build/_sidebar.html.erb:14]: Not yet implemented I guess?

File: app/policies/experiences/build_policy.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+  def create?
+    true
+  end
+
+  def update?
+    raise
+    true
+  end
+end
```
Comment (dmbf29) [inline app/policies/experiences/build_policy.rb:16]: thought you were going to delete this one? :trollface:

File: app/assets/stylesheets/components/_form.scss [hunk 8]
```diff
@@ -0,0 +1,7 @@
+.photo-form-box {
+  padding:50px;
+  border: 0.5px solid gray;
+  box-shadow:0 0 15px 4px rgba(0,0,0,0.06);
+  border-radius: 5px;
```
Comment (dmbf29) [inline app/assets/stylesheets/components/_form.scss:5]: careful on setting something of these arbitrarily by hand. we have a border-radius default set in our boostrap config.

**Review by dmbf29 — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/components/_map.scss (hunk 1 above)
Comment (sashakaverina) [reply app/assets/stylesheets/components/_map.scss:128]: Screen Shot 2022-01-08 at 22 21 19

map and div on it

File: app/views/experiences/new.html.erb (hunk 3 above)
Comment (sashakaverina) [reply app/views/experiences/new.html.erb]: looks like we don't need it anymore. let's delete it

File: app/views/experiences/build/_sidebar.html.erb (hunk 6 above)
Comment (sashakaverina) [reply app/views/experiences/build/_sidebar.html.erb:14]: not yet not yet

File: app/assets/stylesheets/components/_form.scss (hunk 8 above)
Comment (dmbf29) [reply app/assets/stylesheets/components/_form.scss:5]: fixed

**Review by dmbf29 — APPROVED** [review]

**Pull request merged**